modules/percolator/src/main/java/org/elasticsearch/percolator/PercolateQueryBuilder.java

```diff
@@ -281,9 +245,13 @@ protected void doWriteTo(StreamOutput out) throws IOException {
         }
         out.writeString(field);
         out.writeOptionalString(name);
-        out.writeOptionalString(documentType);
+        if (out.getVersion().before(Version.V_8_0_0)) {
+            out.writeOptionalString(MapperService.SINGLE_MAPPING_NAME);
```

Comment:
Related to my comment on the previous revision, I think we should write `null` here. In 7.x, a 'typeless' percolate query is represented by a null documentType and indexedDocumentType.